flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/utils/MongoUtils.java

```diff
@@ -339,8 +349,24 @@ public static <T> MongoCollection<T> collectionFor(
                 .withDocumentClass(documentClass);
     }
 
+    public static String getMongoVersion(MongoDBSourceConfig sourceConfig) {
+        MongoClient client = MongoClientPool.getInstance().getOrCreateMongoClient(sourceConfig);
+        return client.getDatabase("config")
+                .runCommand(new BsonDocument("buildinfo", new BsonString("")))
+                .get("version")
+                .toString();
+    }
+
     public static MongoClient clientFor(MongoDBSourceConfig sourceConfig) {
-        return MongoClientPool.getInstance().getOrCreateMongoClient(sourceConfig);
+        MongoClient client = MongoClientPool.getInstance().getOrCreateMongoClient(sourceConfig);
+        if (sourceConfig.isFullDocPrePostImageEnabled()) {
+            String mongoVersion = getMongoVersion(sourceConfig);
+            if (mongoVersion.charAt(0) < '6') {
+                throw new RuntimeException(
```

Comment:
Could we remove this validation logic to compile phase of the source? thus we can throw the exception earlier, and maybe we can even enable `full.document.pre.post.image` by default for mongo 6+ version after validation.

Reply:
Removed.